lib/pages/assignment/assignment_preview/widgets/assignment_card.dart

```diff
@@ -0,0 +1,128 @@
+  final dynamic question;
+  final int currentQuestionIndex;
+  final int totalQuestions;
+  final int? selectedOption;
+  final Function onOptionSelected;
+  final Function onNextPressed;
+  final Function onUpdatePressed;
+
+  const AssignmentCard({
+    Key? key,
+    required this.question,
+    required this.currentQuestionIndex,
+    required this.totalQuestions,
+    required this.selectedOption,
+    required this.onOptionSelected,
+    required this.onNextPressed,
+    required this.onUpdatePressed,
+  }) : super(key: key);
+
+  Widget _buildOption(String optionText, int optionId, bool isCorrect) {
+    bool isSelected =
+        selectedOption == optionId; // Check if this option is selected
+
+    return Container(
+      margin: const EdgeInsets.only(bottom: 8),
+      decoration: BoxDecoration(
+        border: Border.all(color: isCorrect ? Colors.green : Colors.grey),
+        borderRadius: BorderRadius.circular(12),
+        color: isSelected ? Colors.grey[200] : Colors.white,
+      ),
+      child: ListTile(
+        title: Text(
+          optionText,
+          style: bodyStyle()
+              .copyWith(color: isCorrect ? Colors.green : Colors.grey),
+        ),
+        leading: Radio<int>(
+          value: optionId,
+          groupValue: selectedOption,
+          onChanged: (value) => onOptionSelected(optionId),
+          activeColor: Colors.green,
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Card(
+      elevation: 4,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(12),
+      ),
+      child: Column(
+        children: [
+          Expanded(
+            child: Padding(
+              padding: const EdgeInsets.all(16.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    question['q'] ?? 'No question available',
+                    style: bodyStyle()
+                        .copyWith(fontSize: 18, fontWeight: FontWeight.w500),
+                  ),
+                  const SizedBox(height: 16),
+                  if (question['type'] == 'multi') ...[
+                    Column(
+                      children: question['answers']?.map<Widget>((option) {
+                            final isCorrect = option['is_correct'] == 1;
+                            return _buildOption(option['text'] ?? 'No text',
+                                option['id'] ?? 0, isCorrect);
+                          }).toList() ??
+                          [],
+                    ),
+                  ] else if (question['type'] == 'essay') ...[
+                    Padding(
+                      padding: const EdgeInsets.symmetric(vertical: 16),
+                      child: Text(question['answers']?.toString() ??
+                          'No answer available'),
+                    ),
+                  ],
+                ],
```

Comment:
add 'short answer' and 'true or false' variation for question type

create enum for question type
use case/switch since they are exhaustive, this ensures that all enum values are handled